lib/src/androidTestDebug/java/com/nytimes/android/external/registerlib/GoogleProductResponseTest.java

```diff
@@ -0,0 +1,50 @@
+package com.nytimes.android.external.registerlib;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class GoogleProductResponseTest {
+
+
+    @Test
+    public void testGoodString(){
+
+        String json2 = "{ \"description\": " +
+                "\"Unlimited article access, anytime, anywhere. Save with an annual subscription.\", " +
+                "\"introductoryPrice\": \"$89.99\", \"introductoryPriceCycles\": 1, " +
+                "\"introductoryPriceMicros\": 89990000, " +
+                "\"introductoryPricePeriod\": \"P1Y\", \"price\": \"$129.99\", \"price_amount_micros\": 129990000, " +
+                "\"price_currency_code\": \"USD\", \"productId\": \"com.nytimes.googleplay.bundle.xpass.v1.yearly\", " +
+                "\"subscriptionPeriod\": \"P1Y\", \"title\": \"NYT Basic Access: Annual\", \"type\": \"subs\" }";
+        GoogleProductResponse response = GoogleProductResponse.fromJson(json2);
+        assertNotNull(response);
+    }
+    @Test
+    public void testBadString(){
+        String bad = "{ \"description\": " +
+                "\"Basic access features, plus NYT Crossword, Cooking, and one bonus \\nsubscription.\", " +
+                "\"freeTrialPeriod\": \"P1W\", \"price\": \"$24.99\", \"price_amount_micros\": 24990000, " +
+                "\"price_currency_code\": \"USD\", " +
+                "\"productId\": \"com.nytimes.googleplay.bundle.maxadacr.v1.monthly\", " +
+                "\"subscriptionPeriod\": \"P1M\", " +
+                "\"title\": \"NYT All Access (NYTimes - Latest News)\", \"type\": \"subs\" }";
+
+        GoogleProductResponse response = GoogleProductResponse.fromJson(bad);
+        assertNotNull(response);
```

Comment:
If this is bad JSON, shouldnt the response be null?

Reply:
Previously this was causing an exception to be thrown and not deserializing the data.  
Should maybe make the test name more descriptive of that. Not so much bad, but missing some optional fields.